build/toolchain/linux/BUILD.gn

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+
+import("${build_root}/toolchain/gcc_toolchain.gni")
+
+gcc_toolchain("linux_x64_gcc") {
```

Comment:
We should be able to turn this into templates like

```
template("linux_clang_toolchain") {
  gcc_toolchain("linux_${target_name}_clang") {
    toolchain_args = {
      current_os = "linux"
      current_cpu = invoker.target_name

      is_clang = true
    }
  }
}
```

Then we can use

```
linux_clang_toolchain("x64") {
}
```

Reply:
I think the gains are pretty small. Also, I don't think the target name should affect the behavior of the target.